### PR TITLE
Use typescript, jest, tslint and lerna

### PR DESCRIPTION
**Pull request by benjaminkay93**
Decided to just throw some structure out there, its draft as i havn't actually written and decent test's for the logger, but thats just so I have something to test things build and work nicely together, alot more togo aha

Really useful blog post on setup of all this https://medium.com/grand-parade/monorepo-setup-with-lerna-typescript-babel-7-and-other-part-1-ac60eeccba5f

### CODE REVIEW

File: tsconfig.base.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    // "tsBuildInfoFile": "",                 /* Specify file to store incremental compilation information */
+    // "removeComments": true,                /* Do not emit comments to output. */
+    // "noEmit": true,                        /* Do not emit outputs. */
+    // "importHelpers": true,                 /* Import emit helpers from 'tslib'. */
+    // "downlevelIteration": true,            /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
+    // "isolatedModules": true,               /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */
+
+    /* Strict Type-Checking Options */
+    "strict": true,                           /* Enable all strict type-checking options. */
+    // "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
```
Comment (moschopsuk) [inline tsconfig.base.json]: I think this should be on. We wont get the best experience with typescript if we just set most to `any`.

File: tsconfig.base.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    // "strictNullChecks": true,              /* Enable strict null checks. */
```
Comment (moschopsuk) [inline tsconfig.base.json]: Same for this.

**Comment by benjaminkay93** [comment]
TODO: Make jest interpret ts files again before build

**Comment by thomasslee97** [comment]
All looks sensible to me!

**Pull request merged**